qml/components/StickerPicker.qml

```diff
@@ -87,164 +108,133 @@ Item {
                 flow: GridView.FlowTopToBottom
 
                 model: stickerPickerOverlayItem.recentStickers
+                delegate: stickerComponent
+
+                HorizontalScrollDecorator {}
+
+            }
+        }
+        delegate: Column {
+            id: stickerSetColumn
+
+            property bool isExpanded: false
+            function toggleDisplaySet() {
+                stickerSetColumn.isExpanded = !stickerSetColumn.isExpanded;
+                if (stickerSetColumn.isExpanded) {
+                    stickerSetLoader.myStickerSet = modelData.stickers;
+                }
+            }
 
-                delegate: Item {
-                    width: recentStickersGridView.cellWidth
-                    height: recentStickersGridView.cellHeight
+            spacing: Theme.paddingMedium
+            width: parent.width
 
+            Row {
+                id: stickerSetTitleRow
+                width: parent.width
+                height: Theme.itemSizeMedium + ( 2 * Theme.paddingSmall )
+                spacing: Theme.paddingMedium
+                BackgroundItem {
+                    id: stickerSetToggle
+                    width: parent.width - removeSetButton.width - Theme.paddingMedium * 2
+                    height: parent.height
+
+                    onClicked: {
+                        toggleDisplaySet();
+                    }
                     TDLibThumbnail {
-                        thumbnail: modelData.thumbnail
-                        anchors.fill: parent
+                        id: stickerSetThumbnail
+                        thumbnail: modelData.thumbnail ? modelData.thumbnail : modelData.stickers[0].thumbnail
+                        anchors {
+                            left: parent.left
+                            verticalCenter: parent.verticalCenter
+                            leftMargin: Theme.paddingMedium
+                        }
+                        width: Theme.itemSizeMedium
+                        height: Theme.itemSizeMedium
+                        highlighted: stickerSetToggle.down
                     }
 
                     Label {
-                        font.pixelSize: Theme.fontSizeSmall
-                        anchors.right: parent.right
-                        anchors.bottom: parent.bottom
-                        text: Emoji.emojify(modelData.emoji, font.pixelSize)
+                        id: setTitleText
+                        font.pixelSize: Theme.fontSizeLarge
+                        font.bold: true
+
+                        anchors {
+                            left: stickerSetThumbnail.right
+                            right: expandSetButton.left
+                            verticalCenter: parent.verticalCenter
+                            margins: Theme.paddingSmall
+                        }
+                        maximumLineCount: 1
```

Comment:
I don't think this makes any difference when wrapping if off (which is the default)

Reply:
Yup, I just copy/pasted most stuff from the previous implementation. Good catch!